### PR TITLE
[CryptofeedWebsocketConnector] Add authenticated channels

### PR DESCRIPTION
**Pull request by Herklos**

### CODE REVIEW

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 1]
```diff
@@ -246,14 +260,42 @@ def get_exchange_pair(self, pair) -> str:
...
+        self._create_client_config()
+        self.client = cryptofeed.FeedHandler(config=self.client_config)
+
+    def _create_client_config(self):
+        """
+        Creates cryptofeed config
+        """
+        self.client_config = cryptofeed_config.Config()
+        self.client_config.config.get('log', {})['filename'] = ""
+        self.client_config.config.get('rest', {}).get('log', {})['filename'] = ""
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:277]: why do we need this ?

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 2]
```diff
@@ -246,14 +260,42 @@ def get_exchange_pair(self, pair) -> str:
...
+        self.client_config.config[self.get_feed_name().lower()] = self._get_credentials_config()
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:278]: 👍

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 3]
```diff
@@ -277,26 +319,41 @@ def _start_client(self, should_create_loop=True):
...
+
         self._subscribe_all_pairs_feed()
 
+    def _should_use_authenticated_feeds(self):
+        """
+        :return: True when authenticated feeds shouldn't be added
+        """
+        return self._should_authenticate() and self.exchange.authenticated() and not self.exchange_manager.is_sandboxed
+
+    def _is_supported_channel(self, channel):
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:345]: 👍

File: octobot_trading/personal_data/positions/positions_manager.py [hunk 4]
```diff
@@ -117,7 +117,7 @@ def _select_positions(self, symbol=None):
         """
         if symbol is None:
             return self.positions
-        for position in self.positions:
+        for position in list(self.positions.values()):
```
Comment (GuillaumeDSM) [inline octobot_trading/personal_data/positions/positions_manager.py:120]: we need to load it into a list ?

File: octobot_trading/personal_data/orders/states/fill_order_state.py [hunk 5]
```diff
@@ -100,7 +100,7 @@ async def terminate(self):
             # update position if any
             if self.order.exchange_manager.is_future:
                 await self.order.exchange_manager.exchange_personal_data.positions_manager \
-                    .get_symbol_position(self.order.symbol).update_from_filled_order(self)
+                    .get_symbol_position(self.order.symbol).update_from_filled_order(self.order)
```
Comment (GuillaumeDSM) [inline octobot_trading/personal_data/orders/states/fill_order_state.py:103]: 😂

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py [hunk 6]
```diff
@@ -545,5 +603,53 @@ async def open_interest(self, feed, symbol, open_interest, timestamp, receipt_ti
...
-        pass  # Coingecko only
+    async def order(self, feed, symbol, data, receipt_timestamp):
+        """
+        Cryptofeed order callback
+        :param feed: the feed
+        :param symbol: the order symbol
+        :param data: a dict of miscellaneous data
+        :param receipt_timestamp: received timestamp
+        """
+        await self.push_to_channel(trading_constants.ORDERS_CHANNEL, orders=[self.exchange.parse_order(data)])
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:614]: exchange.parse_order is working here ? :o

**Review by GuillaumeDSM — COMMENTED** [review]
🎉

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py (hunk 6 above)
Comment (Herklos) [reply octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:614]: Yes. I only tested it for Bybit for now and it works :smiley:.

File: octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py (hunk 1 above)
Comment (Herklos) [reply octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py:277]: Cryptofeed logs are currently written in `feedhandler.log`. I disabled logs saving by using an empty file name

**Review by GuillaumeDSM — APPROVED** [review]

**Pull request merged**